pkg/auth/dependency/provider/anonymous/anonymous.go

```diff
@@ -0,0 +1,48 @@
+package anonymous
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/skygeario/skygear-server/pkg/core/db"
+)
+
+const providerPassword string = "anonymous"
+
+type Provider interface {
+	CreatePrincipal(principal Principal) error
+}
+
+type ProviderImpl struct {
+	sqlBuilder  db.SQLBuilder
+	sqlExecutor db.SQLExecutor
+	logger      *logrus.Entry
+}
+
+func NewProvider(builder db.SQLBuilder, executor db.SQLExecutor, logger *logrus.Entry) *ProviderImpl {
+	return &ProviderImpl{
+		sqlBuilder:  builder,
+		sqlExecutor: executor,
+		logger:      logger,
+	}
+}
+
+func (p ProviderImpl) CreatePrincipal(principal Principal) (err error) {
+	// TODO: log
+
+	// Create principal
+	builder := p.sqlBuilder.Insert(p.sqlBuilder.TableName("principal")).Columns(
+		"id",
+		"provider",
+		"user_id",
+	).Values(
+		principal.ID,
+		providerPassword,
```

Comment:
I remember the V1 implementation of "anonymous" user is a user with empty password.

Am I correct? I think in this next refractor should not change it? It will make the data migration before difficult if not make it impossible.

Reply:
> I remember the V1 implementation of "anonymous" user is a user with empty password.

Yes.

But actually what I intended to do when designing the next auth database is the same as what this PR does.